core/database.py

```
@@ -19,13 +21,28 @@ async def init_db():
+            logger.info("Database connection pool created.")
+            await create_schema()
+        except Exception as e:
+            if new_pool is not None:
+                try:
+                    await new_pool.close()
+                except Exception:
+                    logger.exception("Failed to close database pool during init cleanup")
+                finally:
+                    if pool is new_pool:
+                        pool = None
+            logger.error(f"Failed to initialize database: {e}")
```

Ruff (0.15.2)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)